lectures/numpy.md

```diff
@@ -438,6 +438,409 @@ A * B
 (numpy_matrix_multiplication)=
 In particular, `A * B` is *not* the matrix product, it is an element-wise product.
 
+(broadcasting)=
+#### Broadcasting
```

Comment:
Looking at the flow of the lecture, this should be a level 3 heading, so please replace `####` with `###`

Reply:
Please add here

"(This section draws on an excellent discussion of broadcasting provided by [Jake VanderPlas](https://jakevdp.github.io/PythonDataScienceHandbook/02.05-computation-on-arrays-broadcasting.html).)"